addons/account_edi_ubl_cii/models/ir_actions_report.py

```diff
@@ -62,4 +64,39 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                 invoice = self.env['account.move'].browse(res_id)
                 self._add_pdf_into_invoice_xml(invoice, stream_data)
 
+            # If Factur-X isn't already generated, generate and embed it inside the PDF
+            if len(res_ids) == 1:
```

Comment:
Why this limitation?
Why not iterate over the invoices?

And shouldn't it be done before the call to `_add_pdf_into_invoice_xml`?

Reply:
printing several invoices will produce a single PDF with all the invoices: we cannot generate Factur-X for this scenario.
![image](https://user-images.githubusercontent.com/60702485/193831237-e55ed2d7-b078-4fdb-a89e-26cb95f22155.png)

Reply:
`_add_pdf_into_invoice_xml` will embed the PDF inside the xml. For Factur-X, we embed the xml inside the PDF. We never do both for a given format (it would be useless...).